krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -46,13 +49,29 @@ type Interface interface {
 	// SetNetworkInstanceName sets the name of the networkInstance in the spec
 	// returns an error when the set fails
 	SetNetworkInstanceName(s string) error
-	// SetInterfaceSpec sets the spec attributes in the kubeobject
-	SetInterfaceSpec(*nephioreqv1alpha1.InterfaceSpec) error
+	// SetSpec sets the spec attributes in the kubeobject
+	SetSpec(*nephioreqv1alpha1.InterfaceSpec) error
+
+	DeleteAttachmentType() error
+
+	DeleteCNIType() error
 }
 
 // New creates a new parser interface
 // It expects a raw byte slice as input representing the serialized yaml file
-func New(b []byte) (Interface, error) {
+func New(x any) (Interface, error) {
```

Comment:
Alternatively you can have `NewFromYAML([]byte)` and `NewFromGoStruct(*nephioreqv1alpha1.Interface)`. Is there a reason to use the same name?

Reply:
yes that works as well. I agree it is better to be explicit. Will change.

Reply:
fixed